{
  "name": "sls_app_etl_at_error_monitor",
  "displayName": "数据加工异常报错监控",
  "type": "Alert",
  "state": "Enabled",
  "schedule": {
    "type": "FixedRate",
    "interval": "5m"
  },
  "configuration": {
    "version": "2.0",
    "type": "default",
    "dashboard": "internal-alert-analysis",
    "queryList": [
      {
        "storeType": "log",
        "project": "{{default.project}}",
        "store": "internal-etl-log",
        "query": "(__topic__:  __etl-log-status__ and \"logging.levelname\": ERROR) | select \"__tag__:__schedule_id__\" as job_id, arbitrary(\"__tag__:__job_name__\") as job_name, arbitrary(\"etl_context.logstore\") as logstore, count(*) as cnt, array_agg(distinct etl_context) as etl_context, array_agg(distinct reason) as reason from log where regexp_like(\"__tag__:__schedule_id__\", '.*') group by job_id limit 10000",
        "timeSpanType": "Truncated",
        "start": "-5m",
        "end": "now"
      }
    ],
    "groupConfiguration": {
      "type": "custom",
      "fields": [
        "job_id"
      ]
    },
    "joinConfigurations": [],
    "severityConfigurations": [
      {
        "severity": 8,
        "evalCondition": {
          "condition": "",
          "countCondition": ""
        }
      }
    ],
    "labels": [],
    "annotations": [
      {
        "key": "desc",
        "value": "过去5分钟内，源logstore ${logstore}下的数据加工作业(作业ID:${job_id}, 作业名称:${job_name})产生了${cnt}条报错日志，etl_context: ${etl_context}, reason: ${reason}。更多报错细节请前往加工日志库internal-etl-log查看。"
      },
      {
        "key": "title",
        "value": "数据加工异常报错监控"
      },
      {
        "key": "__drill_down_query__",
        "value": "__topic__:  __etl-log-status__ and \"logging.levelname\": ERROR and __tag__:__schedule_id__: ${job_id}"
      }
    ],
    "autoAnnotation": true,
    "sendResolved": false,
    "threshold": 1,
    "noDataFire": false,
    "noDataSeverity": 6
  }
}